basic_nodes/src/main/java/club/mondaylunch/gatos/basicnodes/HTTPRequestNodeType.java

```diff
@@ -0,0 +1,145 @@
+        var method = DataBox.get(settings, "method", DataType.STRING).orElse("").toUpperCase();
+        var body = DataBox.get(inputs, "body", DataType.STRING).orElse("");
+        
+        // get uri by checking url and method
+        URI uri = this.getURI(url, method);
+
+        if (uri == null) {
+            return this.handleInvalidReturns();
+        }
+
+        // the url and method are valid so we create the request
+        HttpRequest request = this.createRequest(method, uri, body);
+        double statusCode;
+        String responseBody;
+    
+        // send the request
+        try {
+            HttpResponse<String> response = this.sendRequest(request);
+            statusCode = response.statusCode();
+            responseBody = response.body();
+        } catch (IOException | InterruptedException e) {
+            // if a problem happens just return nothing
+            return this.handleInvalidReturns();
+        }
+
+        return Map.of(
+            "StatusCode", CompletableFuture.completedFuture(DataType.NUMBER.create(statusCode)),
+            "responseText", CompletableFuture.completedFuture(DataType.STRING.create(responseBody))
+        );
+    }
+
+    private Map<String, CompletableFuture<DataBox<?>>> handleInvalidReturns() {
+        return Map.of(
+            "StatusCode", CompletableFuture.completedFuture(DataType.NUMBER.create(404.0)),
+            "responseText", CompletableFuture.completedFuture(DataType.STRING.create("URL or method are incorrect"))
+        );
+    }
+
+    private URI getURI(String url, String method) {
+        URI uri = null;
+        if (this.validateMethod(method)) {
+            try {
+                uri = new URI(url);
+            } catch (URISyntaxException e) {
+                uri = null;
+            }
+        }
+        return uri;
+    }
+
+    private boolean validateMethod(String method) {
+        return Methods.isMethod(method);
+    }
+
+    private HttpRequest createRequest(String method, URI uri, String body) {
+        // note: this method is called after validating the method parameter so the switch will execute
+        // no null will be returned 
+        Builder builder = HttpRequest.newBuilder();
+        HttpRequest request = null;
+        switch (method) {
+            case "GET": request = builder.uri(uri).GET().build(); break;
+            case "POST": request = builder.uri(uri).POST(BodyPublishers.ofString(body)).build(); break;
+            case "PUT": request = builder.uri(uri).PUT(BodyPublishers.ofString(body)).build(); break;
+            case "DELETE": request = builder.uri(uri).DELETE().build(); break;
+        }
+        return request;
+    }
+
+    private HttpResponse<String> sendRequest(HttpRequest request) throws IOException, InterruptedException {
+        HttpClient httpClient = HttpClient.newHttpClient();
+        HttpResponse<String> response = httpClient.send(request, BodyHandlers.ofString());
```

Comment:
HttpClient has a sendAsync method that returns a CompletableFuture that you probably want to take advantage of.